src/css/showHide.js

```diff
@@ -19,23 +36,30 @@ function showHide( elements, show ) {
 
 		display = elem.style.display;
 		if ( show ) {
-			if ( display === "none" ) {
 
-				// Restore a pre-hide() value if we have one
-				values[ index ] = dataPriv.get( elem, "display" ) || "";
+			// Since we force visibility upon cascade-hidden elements, an immediate (and slow)
+			// check is required in this first loop unless we have a nonempty display value (either
+			// inline or about-to-be-restored)
+			if ( display === "none" ) {
+				values[ index ] = dataPriv.get( elem, "display" ) || null;
+				if ( !values[ index ] ) {
+					elem.style.display = "";
+				}
+			}
+			if ( elem.style.display === "" && jQuery.css( elem, "display" ) === "none" ) {
+				values[ index ] = defaultDisplayMap[ elem.nodeName ] || setDefaultDisplay( elem );
```

Comment:
So `setDefaultDisplay` is a getter? I would suggest to do write it like 
```js
values[ index ] = getDefaultDisplay(elem);
```

And incapsulate everything else